src/mender-update/daemon/context.cpp

```diff
@@ -351,6 +351,68 @@ error::Error Context::SaveDeploymentStateData(StateData &state_data) {
 		return SaveDeploymentStateData(txn, state_data);
 	});
 }
+static error::Error UnmarshalJsonStateDataVersion1(const json::Json &json, StateData &state_data) {
+#define SetOrReturnIfError(dst, expr) \
+	if (!expr) {                      \
+		return expr.error();          \
+	}                                 \
+	dst = expr.value()
+
+#define DefaultOrSetOrReturnIfError(dst, expr, def)                            \
+	if (!expr) {                                                               \
+		if (expr.error().code == kv_db::MakeError(kv_db::KeyError, "").code) { \
+			dst = def;                                                         \
+		} else {                                                               \
+			return expr.error();                                               \
+		}                                                                      \
+	} else {                                                                   \
+		dst = expr.value();                                                    \
+	}
```

Comment:
Both of these macros are also defined in the original `error::Error UnmarshalJsonStateData`. Could we move them now to helper functions instead?

Reply:
Yeh, probably should - fixing``